pkg/resource/function/hooks.go

```diff
@@ -614,63 +621,140 @@ func (rm *resourceManager) setResourceAdditionalFields(
+					}
 				}
-			}
-			if getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess != nil {
-				if getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess.Destination != nil {
-					ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination = getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess.Destination
+				if getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess != nil {
+					if getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess.Destination != nil {
+						ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination = getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess.Destination
+					}
 				}
+			} else {
+				ko.Spec.FunctionEventInvokeConfig.DestinationConfig = nil
+			}
+			if getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds != nil {
+				ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds = getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds
+			} else {
+				ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds = nil
+			}
+			if getFunctionEventInvokeConfigOutput.DestinationConfig != nil {
+				ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts = getFunctionEventInvokeConfigOutput.MaximumRetryAttempts
+			} else {
+				ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts = nil
 			}
 		} else {
-			ko.Spec.FunctionEventInvokeConfig.DestinationConfig = nil
-		}
-		if getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds != nil {
-			ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds = getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds
-		} else {
-			ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds = nil
-		}
-		if getFunctionEventInvokeConfigOutput.DestinationConfig != nil {
-			ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts = getFunctionEventInvokeConfigOutput.MaximumRetryAttempts
-		} else {
-			ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts = nil
+			cloudFunctionEventInvokeConfig := &svcapitypes.PutFunctionEventInvokeConfigInput{}
+			cloudFunctionEventInvokeConfig.DestinationConfig = &svcapitypes.DestinationConfig{}
+			cloudFunctionEventInvokeConfig.DestinationConfig.OnFailure = &svcapitypes.OnFailure{}
+			cloudFunctionEventInvokeConfig.DestinationConfig.OnSuccess = &svcapitypes.OnSuccess{}
+			cloudFunctionEventInvokeConfig.DestinationConfig.OnFailure.Destination = getFunctionEventInvokeConfigOutput.DestinationConfig.OnFailure.Destination
+			cloudFunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination = getFunctionEventInvokeConfigOutput.DestinationConfig.OnSuccess.Destination
+			cloudFunctionEventInvokeConfig.MaximumEventAgeInSeconds = getFunctionEventInvokeConfigOutput.MaximumEventAgeInSeconds
+			cloudFunctionEventInvokeConfig.MaximumRetryAttempts = getFunctionEventInvokeConfigOutput.MaximumRetryAttempts
+			ko.Spec.FunctionEventInvokeConfig = cloudFunctionEventInvokeConfig
```

Comment:
Well here I am not assigning `ko.Spec.FunctionEventInvokeConfig` to nil, instead creating the `FunctionEventInvokeConfig` object and assigning it to the values returned from `GetFunctionEventInvokeConfig` API call. I am creating the whole `FunctionEventInvokeConfig` object because this is for the case where the desired spec for `FunctionEventInvokeConfig` is nil and `Spec.ko.FunctionEventInvokeConfig` doesn't exist. So we are creating the whole object.
To create the `FunctionEventInvokeConfig` I need to define every object properly, since the `DestinationConfig` field is highly nested, I had to define their respective type for each following fields. This is the reason behind this multiple lines of code. Tbh, I am not okay with so many lines either but I don't see other way around.